File: 2024_VIT Bhopal_Lions should Roar/Technical Task 3/Question 1.c
/*Problem statement:
A chocolate factory is packing chocolates into packets. The chocolate packets here represent an array arr[] of N number of integer values. The task is to find the empty packets(0) of chocolate and push it to the end of the conveyor belt(array).

For example:
N = 7 and arr = [4, 5, 0, 1, 9, 0, 5, 0]
There are 3 empty packets in the given set. These 3 empty packets represented as 0 should be pushed toward the end of the array.

Example 1
Input
7
4 5 0 1 9 0 5

Output
4 5 1 9 5 0 0 

Example 2
Input
6
6 0 1 8 0 2

Output
6 1 8 2 0 0

Input format
The first line of input contains an integer n denoting the size of an array
The next space-separated integers denote the elements of an array

Output format
The output prints an integer array separated by space.

Sample testcases
Input 1
7
4 5 0 1 9 0 5
  
Output 1
4 5 1 9 5 0 0 

Input 2
6
6 0 1 8 0 2
  
Output 2
6 1 8 2 0 0

Code:-*/
/*Solution:-*/
#include<iostream>
using namespace std;
void pushZeroToEnd(int arr[], int n)
{
    int count = {0};
    for (int i = 0; i < n; i++)
    if (arr[i] != 0)
    arr[count++] = arr[i];
    while(count < n)
    arr[count++] = 0;
}

int main()
{
    int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++)
    cin >> arr[i];
    pushZeroToEnd(arr, n);
    for (int i = 0; i < n; i++)
    cout << arr[i] << " ";
    return 0;
}

File: 2024_VIT Bhopal_Lions should Roar/Technical Task 1/Question 1.py
"""Problem statement:
Write a program to check whether the given matrix is square or not.

Input format
The first line of the input consists of the number of rows and columns.
The next input is the matrix.

Output format
The output prints the matrix.
Then whether the matrix is square or not.

Refer sample output.
Sample testcases
Input 1
3 3
1 2 3
4 5 6
7 8 9

Output 1
1 2 3 
4 5 6 
7 8 9 
The entered array is a square matrix.

Input 2
3 4
1 2 3 4
5 6 7 8
9 10 11 12

Output 2
1 2 3 4 
5 6 7 8 
9 10 11 12 
The entered array is not a square matrix.

Code:-"""

num_rows, num_cols = map(int, input().split())
matrix = []
for i in range(num_rows):
    matrix.append(list(map(int,input().split())))
for row in matrix:
    print(*row)
if num_rows == num_cols:
    print("The entered array is a square matrix.")
else:
    print("The entered array is not a square matrix.")
